Packs/CofenseTriage/Integrations/CofenseTriage/CofenseTriage.py

```diff
@@ -51,145 +25,248 @@
     'category_id',
     'match_priority',
     'tags',
-    'email_attachments'
+    'email_attachments',
 ]
 
 
-# HELPER FUNCTIONS #
-def snake_to_camel_keys(snake_list: List[Dict]) -> List[Dict]:
-    def snake_to_camel(snake_str) -> str:
-        if snake_str == 'id':
-            return 'ID'
-        components = snake_str.split('_')
-        return ''.join(x.title() for x in components)
-
-    return [{snake_to_camel(k): v for k, v in snake_d.items()} for snake_d in snake_list]
-
-
-def split_snake(string: str) -> str:
-    return string.replace("_", " ").title()
+class TriageInstance:
+    def __init__(self, *, host, token, user, disable_tls_verification=False):
+        self.host = host
+        self.token = token
+        self.user = user
+        self.disable_tls_verification = disable_tls_verification
 
+    def request(self, endpoint, params=None, body=None, raw_response=False):
+        """
+        Make a request to the configured Triage instance and return the result.
+        """
+        # TODO automatic rate-limiting
+        response = requests.get(
+            self.api_url(endpoint),
+            headers={
+                "Authorization": f"Token token={self.user}:{self.token}",
+                "Accept": "application/json",
+            },
+            params=params,
+            data=body,
+            verify=not self.disable_tls_verification,
+        )
 
-# MAIN FUNCTIONS #
-def http_request(url_suffix: str, params=None, body=None, raw_response=False) -> Any:
-    """
-    Generic request to Cofense Triage. Client applications can make 25 requests to Cofense Triage
-    within a five-minute interval using the Cofense Triage API.
-    """
-    response = requests.get(
-        BASE_URL + url_suffix,
-        headers=HEADERS,
-        params=params,
-        data=body,
-        verify=USE_SSL,
-    )
-    try:
         if not response.ok:
-            return_error(f'Call to Cofense Triage failed [{response.status_code}] - [{response.text}]')
+            return return_error(
+                f"Call to Cofense Triage failed ({response.status_code}): {response.text}"
+            )
 
-        elif response.status_code == 206:  # 206 indicates Partial Content, reason will be in the warning header
+        if response.status_code == 206:
+            # 206 indicates Partial Content. The reason will be in the warning header.
             demisto.debug(str(response.headers))
 
         if raw_response:
+            # TODO refactor to get rid of this?
```

Comment:
why? I see that it is being used in some of the commands.

Reply:
It possibly could be made unnecessary with more refactoring in the future.